core/sdk.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2023 Mia srl
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/rond-authz/rond/internal/metrics"
+	"github.com/rond-authz/rond/openapi"
+	"github.com/rond-authz/rond/types"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+	"github.com/uptrace/bunrouter"
+)
+
+type PolicyResult struct {
+	QueryToProxy []byte
+	Allowed      bool
+}
+
+// Warning: This interface is experimental, and it could change with breaking also in rond patches.
+// Does not use outside this repository until it is not ready.
+type SDK interface {
+	Metrics() metrics.Metrics
+
+	FindEvaluator(logger *logrus.Entry, method, path string) (SDKEvaluator, error)
+	EvaluatorFromConfig(logger *logrus.Entry, config openapi.RondConfig) SDKEvaluator
+}
+
+// Warning: This interface is experimental, and it could change with breaking also in rond patches.
+// Does not use outside this repository until it is not ready.
+type SDKEvaluator interface {
```

Comment:
The final package will be `sdk` or `rond` ? I think the final name for this interface won't be `SDKEvaluator` but only `Evaluator` what do you think? I think that the Evaluator name is already taken within this package but let's prepare in changing it, do you agree?

Reply:
Yes, probably this will change. I expect an external sdk package, but it's not possible at the moment